packages/lib/algolia.cfc

```diff
@@ -731,6 +741,16 @@ component {
 		for (row in qContent) {
 			for (indexName in indexableTypes[qContent.typename]) {
 				if (qContent.operation eq "updated") {
+					
+				// Approved only - draft records do not get updated to approved; new records added to index each time object goes to draft
+				if (StructKeyExists(arguments.stObject, 'status') AND arguments.stObject['status'] != 'approved') {
+					stResult["typename"] = arguments.stObject.typename;
+					stResult["count"] = 0;
+					stResult["builtToDate"] = builtToDate;	
+
+					return 	stResult;
```

Comment:
Return inside a loop - I think this needs to be changed. The other items in the loop still need to be handled.

Reply:
replaced with break

```
		for (row in qContent) {
			for (indexName in indexableTypes[qContent.typename]) {
				if (qContent.operation eq "updated") {
					
				// Approved only - draft records do not get updated to approved; new records added to index each time object goes to draft
				if (StructKeyExists(arguments.stObject, 'status') AND arguments.stObject['status'] != 'approved') {
					break;
				}
```